### DISCUSSION TITLE
Monkey365 getting stucked into Linux Docker Container

### DISCUSSION THREAD

**Discussion by brenosilva** [silverhack/monkey365#33] [Q&A]
Hi @silverhack ,

I was able to run monkey365 on a Windows box, now i´m trying  to execute it from a Linux docker container.
It is executing but in the end for some reason is being get stucked here:

`clientApplicationId           : *****************************
TenantId                      : ***********************************
exportTo                      : {JSON}
LocalizedDataParams           : {FileName, BindingVariable, BaseDirectory}
BatchSleep                    : 4000
BatchSize                     : 200
MaxQueue                      : 2
nestedRunspaceMaxThreads      : 1
outDir                        : /home/monkey365/monkey365/monkey-reports
excludePlugins                :
excludedResources             :`

The script is not being terminated and looks like not saving all files at outDir
Do you have any idea what could be the reason ?

Thanks

**Comment by brenosilva** [comment]
I can see some errors like this:
Write-Error: /home/monkey365/monkey365/core/modules/monkeyjob/public/Get-JobStatus.ps1:103
Line |
 103 |  …             Write-Error ($script:messages.ErrorOnThreadEndInvoke -f $ …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | [20:14:32:968] - [Get-JobStatus] - Error on Thread EndInvoke : Exception calling "EndInvoke" with "1" argument(s): "Object reference not set to an instance of an object." - error - 0c82b13fcb55 -

Write-Error: /home/monkey365/monkey365/core/modules/monkeyjob/public/Get-JobStatus.ps1:104
Line |
 104 |  …             Write-Error ($script:messages.DetailedEndInvokeErrorMessa …
     |                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
     | [20:14:32:973] - [Get-JobStatus] - It was received while processing Object Get-MonkeyTeamsGuestMessagingConfiguration. Job ID: fish-jupiter-fanta-fourteen - error - 0c82b13fcb55 -


Linux is an Ubuntu 22 box

**Reply by silverhack** [reply]
Hi @brenosilva, hope you're doing well, 

There are multiple reasons for that to happen (e.g. Connection timeout, lack of privileges, lack of authentication support etc..). Please also note that not all applications are supporting all authentication flows. For example in Microsoft Teams there is no "official" support for client credential flow. 

I've created two new issues to track the "[warning fatigue](https://github.com/silverhack/monkey365/issues/34)" and to add some improvements in the [plugin architecture](https://github.com/silverhack/monkey365/issues/28).

Hope that helps,
